contracts/infrastructure/dapp/CurveFilter.sol

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2021  Argent Labs Ltd. <https://argent.xyz>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.s
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.12;
+
+import "./BaseFilter.sol";
+
+contract CurveFilter is BaseFilter {
+  bytes4 private constant EXCHANGE = bytes4(keccak256("exchange(int128,int128,uint256,uint256)"));
+  bytes4 private constant ERC20_APPROVE = bytes4(keccak256("approve(address,uint256)"));
+
+  function isValid(address /*_wallet*/, address _spender, address _to, bytes calldata _data) external view override returns (bool) {
+    if (_data.length < 4) {
+        return false;
+    }
+    bytes4 methodId = getMethod(_data);
+    if(_spender == _to) {
+        return (methodId == EXCHANGE);
```

Comment:
Not sure if it is necessary, but maybe we should check that we are swapping stETH for ETH, and not the other way around. Can't imagine how it can be harmful, but our intention with this filter is to only swap stETH for ETH so we might as well enforce it in the code.

If we decide to do so (e.g. by passing the stETH address in the constructor) then we could also check that `_to == stETH` when the method is ERC20_APPROVE).